pytest_plugins/factory_collection.py

```diff
@@ -0,0 +1,41 @@
+import re
+from inspect import getmembers
+from inspect import isfunction
+
+from robottelo.config import settings
+
+
+def pytest_configure(config):
+    """Register markers related to testimony tokens"""
+    marker = 'factory_instance: Test uses a fresh satellite or Capsule instance deployed by broker'
+    config.addinivalue_line("markers", marker)
+
+
+def pytest_collection_modifyitems(session, items, config):
+    from pytest_fixtures.core import factory
+
+    deselected = []
+    selected = []
+    include_factory_upgrades = config.getvalue('include_factory_upgrades')
+    factory_fixture_names = [m[0] for m in getmembers(factory, isfunction)]
+    for item in items:
+        has_factoryfixture = set(item.fixturenames).intersection(set(factory_fixture_names))
+        if has_factoryfixture:
+            item.add_marker('factory_instance')
+        if 'upgrade' in config.option.markexpr and not config.getvalue('include_factory_upgrades'):
+            if all([item.get_closest_marker('upgrade'), has_factoryfixture]) or item.get_closest_marker('destructive'):
```

Comment:
Can we combine these together? Or was this done to make it easier to read?

Reply:
Earlier I wanted to have two conditions - in the first condition one with an upgrade mark and another condition with a destructive mark. But I need not to have it and they combiningly become the one condition on line 26.  I didnt rethink of making it one by combining it with line 25.

But I would like to keep that as is, as both of these conditions has is its own individuality .